### ISSUE TITLE
Log trials responsible for the creation of another trial

### ISSUE THREAD

**Issue by bouthilx** [Epistimio/orion#87]
In the context of experiment version control trials generated by an algorithm may be biased by trials coming from parent or children experiments. To help identify such potential biases we should log in a new trial which trials was part of the algorithm's history at the time it generated it.

We could represent the log as a simple list of trial ids. However if there is thousands of trials then the size of this list will significantly out-weight the rest of the trial. This could use a significant amount of memory and slow down queries when the list of ids is selected.

Another solution is to represent the log as a tree of trial ids. A trial (1) would point to trial (2) which point to trial (3), meaning that trial (1) was created with both trial (2) and (3) in algorithm's history. Worst case scenario is a flat tree, which brings back to the list of ids, but it does not seam like a very likely situation, unless someone sample 1000 trials before starting executing them.

**Comment by DatCorno** [comment]
Isn't this already done?

**Comment by bouthilx** [comment]
Waiting for #152
